### PR TITLE
[INLONG-8358][Sort] Add kafka connector on flink 1.15

### LINKED ISSUE #8358: [Feature][Sort] Add Kafka connector on Flink 1.15
### Description
Parent issue: https://github.com/apache/inlong/issues/8336

Add kafka connector on flink 1.15

### Use case

Add kafka connector on flink 1.15

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hnrainll**
### Prepare a Pull Request

- Fixes #8358 

### Motivation

Add kafka connector on flink 1.15

### Modifications

Add kafka connector on flink 1.15

### CODE REVIEW

**Comment by hnrainll** [comment]
OK

**Comment by hnrainll** [comment]
> Please add a end to end test under sort-end-to-end-tests to test the code thx ~

ok

**Comment by dockerzhang** [comment]
@hnrainll please fix the conflicts, thanks.

**Comment by hnrainll** [comment]
> @hnrainll please fix the conflicts, thanks.

OK

**Comment by EMsnap** [comment]
Sorry but there should be It for this feature, any problem with implementing it ?

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/KafkaE2EITCase.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            return PlaceholderResolver.getDefaultResolver().resolveByMap(
+                    new String(Files.readAllBytes(file), StandardCharsets.UTF_8),
+                    properties);
+        } catch (IOException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Test flink sql mysql cdc to hive
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/KafkaE2EITCase.java:194]: maybe not hive here

**Review by vernedeng — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by e-mhui — APPROVED** [review]

**Pull request merged**